app/views/requests/comments/_list.html.erb

```diff
@@ -1,10 +1,10 @@
 
 <ul id="request_annotations_<%= @request.id %>" class="list-group">
-  <% if @request.comments.size > 0 %>
+  <% if @comments.present? %>
     <% @comments.each do |comment| %>
```

Comment:
The actual bug was looking at the comments on request specifically, not the aggregate in @comments.